server/application/rest/service_shift.py

```diff
@@ -0,0 +1,30 @@
+"""
+This module handles service shift operations.
+"""
+from flask import Blueprint, request, jsonify
+from server.use_cases.add_service_shifts import shift_add_use_case
+from server.repository.mongorepo import MongoRepo
+from server.domains.service_shift import ServiceShift
+
+service_shift_bp = Blueprint('service_shift', __name__)
+
+repo = MongoRepo()
+
+@service_shift_bp.route('/service_shift', methods=['POST'])
+def create_service_shift():
+    """
+    shift creation requests here
+    """
+    data = request.get_json()
+    new_shift = ServiceShift.from_dict(data)
+    existing_shifts = repo.get_shifts_for_volunteer(data.get('user_id'))
+    response = shift_add_use_case(repo, new_shift, existing_shifts)
```

Comment:
You should handle conflicts with existing shifts inside the shift_add_use_case. It can call the shift_list_use_case to get all shifts for the shelter specified in the new_shift, and check for overlaps in time with the new_shift.
```suggestion
    response = shift_add_use_case(repo, new_shift)
```